src/lib/converter/nodes/constructor.ts

```diff
@@ -42,7 +42,10 @@ export class ConstructorConverter extends ConverterNodeComponent<ts.ConstructorD
             if (!hasBody || !method.signatures) {
                 var name = 'new ' + parent.name;
                 var signature = createSignature(context, node, name, ReflectionKind.ConstructorSignature);
-                signature.type = new ReferenceType(parent.name, ReferenceType.SYMBOL_ID_RESOLVED, parent);
+                 // If no return type defined, then use the parent one.
+                if (!node.type) {
```

Comment:
When there is no return type ( ex:`new();` ) then `signature.type == any` and is override with the `parent.name` by `new ReferenceType()`. Still it is possible to define an `any` return type even if not recommended (ex: `new() : any;`).
So checking `signature.type` is not concluant. `node.type` is not defined if no return type written and match the use case.
In my case, the execution is going through the [first extractSignatureType return](https://github.com/TypeStrong/typedoc/blob/22ccb660ae6eb5b3e136dc277ecc1d62fb04d016/src/lib/converter/factories/signature.ts#L58). No possibility to handle the use case here.